File: scheduler/core/src/main/java/io/orkes/conductor/scheduler/listener/ScheduleChangeListenerStub.java
package io.orkes.conductor.scheduler.listener;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import io.orkes.conductor.scheduler.model.WorkflowSchedule;

/** Stub listener default implementation. Logs each schedule change at debug level. */
public class ScheduleChangeListenerStub implements ScheduleChangeListener {

    private static final Logger LOGGER = LoggerFactory.getLogger(ScheduleChangeListenerStub.class);

    @Override
    public void onScheduleRegistered(WorkflowSchedule schedule) {
        LOGGER.debug("Schedule {} registered", schedule.getName());
    }

    @Override
    public void onScheduleUpdated(WorkflowSchedule schedule) {
        LOGGER.debug("Schedule {} updated", schedule.getName());
    }

    @Override
    public void onScheduleDeleted(String name) {
        LOGGER.debug("Schedule {} deleted", name);
    }

    @Override
    public void onSchedulePaused(WorkflowSchedule schedule) {
        LOGGER.debug("Schedule {} paused", schedule.getName());
    }

    @Override
    public void onScheduleResumed(WorkflowSchedule schedule) {
        LOGGER.debug("Schedule {} resumed", schedule.getName());
    }
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/scheduler/listener/ScheduleChangeListenerStub.ts
import { ScheduleChangeListener } from './ScheduleChangeListener';
import { WorkflowSchedule } from '../model/WorkflowSchedule';

/** Stub listener default implementation. Logs each schedule change at debug level. */
export class ScheduleChangeListenerStub implements ScheduleChangeListener {

    // Using console for logger placeholder
    private static readonly LOGGER = console;

    public onScheduleRegistered(schedule: WorkflowSchedule): void {
        ScheduleChangeListenerStub.LOGGER.debug(`Schedule ${schedule.getName()} registered`);
    }

    public onScheduleUpdated(schedule: WorkflowSchedule): void {
        ScheduleChangeListenerStub.LOGGER.debug(`Schedule ${schedule.getName()} updated`);
    }

    public onScheduleDeleted(name: string): void {
        ScheduleChangeListenerStub.LOGGER.debug(`Schedule ${name} deleted`);
    }

    public onSchedulePaused(schedule: WorkflowSchedule): void {
        ScheduleChangeListenerStub.LOGGER.debug(`Schedule ${schedule.getName()} paused`);
    }

    public onScheduleResumed(schedule: WorkflowSchedule): void {
        ScheduleChangeListenerStub.LOGGER.debug(`Schedule ${schedule.getName()} resumed`);
    }
}
